apps/precinct-scanner/src/AppRoot.tsx

```diff
@@ -756,9 +763,10 @@ const AppRoot: React.FC<Props> = ({ hardware, card }) => {
     }
   }, [startUsbStatusPolling, startCardShortValueReadPolling])
 
-  const updatePrecinctId = (precinctId: string) => {
+  const updatePrecinctId = useCallback(async (precinctId: string) => {
     dispatchAppState({ type: 'updatePrecinctId', precinctId })
-  }
+    await config.setCurrentPrecinctId(precinctId)
+  }, [])
```

Comment:
should dispatchAppState be a dependency here, I see you added it in other places?